lib/CL/devices/hsa/pocl-hsa.c

```diff
@@ -333,18 +511,23 @@ pocl_hsa_run
             POCL_ABORT("pocl-hsa: too many kernel arguments!");
           /* Assuming the pointers are 64b (or actually the same as in
              host) due to HSA. TODO: the 32b profile. */
+
+          //Alignment Check
+//          if(args_offset%8 !=0 )
+//        	  args_offset = (args_offset+8)/8;
           if (al->value == NULL)
             {
-              args->kernel_args[args_offset] = 0;
-              args->kernel_args[args_offset + 1] = 0;
+              uint64_t temp = 0;
+        	  memcpy(args->kernel_args + args_offset, &temp, sizeof(uint64_t));
```

Comment:
It is inspired by CLOC samples in HSA Foundation github.
This makes scalar arguments not limit to 32-bit valuables,
and it affects alignments afterwards, which may not align to 4 bytes.